enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,348 @@
+  - the release image fixtures referencing x86 imagestreams. these would need to be changed to manifestlist references or there would need
+  to be imagestreams added for individual architectures and they would be picked based on the node architecture.
+  - references to some images like cluster-api-provider-aws, apiserver-network-proxy etc.. are x86. these would need to change to provide the images
+  for other architectures too. These images can be overridden through annotations today.
+- Another option would be to mandate an x86 worker node be present for Hypershift and nodeselect the x86 node for the hosted cluster control plane
+deployment.
+- When creating the nodepool, the defaults for instances types and AMIs are x86 specific. The instance types can be overridden but not the AMIs.
+An option needs to be added to the create cluster command to override AMIs.
+
+### Console
+
+- As of today console filters content based on control plane architecture. This needs to be entirely removed or made configurable for
+heterogeneous clusters. The operator should be able to filter out the set of OLM operators that, if installed will be able to find
+worker nodes to run on. This means that it has to display only operators that can be supported by architectures that the workers are
+running on.
+- Console would also have to tell OLM for each operator which architectures to nodeselect on by creating a subscription object and adding a
+podAffinity rule with nodeselectorterms. The nodeselectorterms will be defaulted with the architectures that the operator supports, but would
+also have a provision for users to specify the architectures they want (from the set of supported architectures).
+
+### Install
+
+For Phase 1, the way to add more workers to the cluster is to handcraft a machineset manifest file for a worker node and specify the architecture
+specific details like instance type, image references and deploy the machineset to get a worker node. This can be pseudo automated further by creating
+the manifests using `openshift-install create manifests` and modifying or adding worker manifest files. Eventually the installer should support being
+able to declare differing compute and control plane sections in the install-config mainly allowing architectures to be different. In doing so it would
+also generate appropriate machineset and machineconfig files tailored to the specific architecture. The machineset file would contain the instance
+types and region corresponding to the architecture chosen for the workers.
+
+### Open Questions
+
+- What would be the method to provide recommendation updates when moving from a homogeneous graph to a heterogeneous graph ?
+
+### Test Plan
+
+Testing for Phase 1:
+- QE would need to test heterogeneous nightly payloads before the entry is added to Cincinnati. This implies that the heterogeneous payloads will be
+released a bit later than the homogeneous payloads.
```

Comment:
"released" is a bit fuzzy.  We currently build all the homogeneous arch releases and then add them to the `candidate` channels.  But the release isn't GA until errata is published, usually a week later.  I don't think anyone will be particularly worried about a small delay before entry in the `candidate` channels.  Especially for a TechPreview phase 1.  But it might help make this clearer if you replace "released" with "pushed to Quay" or "added to `candidate` channels" or whatever point in the process you're focusing on here.

Reply:
yes this is true but the intent here was that we did not want to hold up homogeneous releases if the heterogeneous testing uncovered some issues which would make it  release blocking. i will change the wording here.